### PR TITLE
feat(auth): mock idp

### LINKED ISSUE #379: Implement the frontend for consent screen
UI to show grant details that user can consent to, adjust, or deny.

### LINKED ISSUE #488: Auth/Identity Integration
* [ ] Consent Screen
  * [ ] UI
  * [ ] Auth Server Integration
* [ ] Allow user selection

### PR DESCRIPTION
**Pull request by dclipp**
## Changes proposed in this pull request

- create a mock IDP for testing purposes
- sub-tasks:
    - [X] basic UI
    - [x] dynamic configuration
    - [x] scaffold API calls
    - [x] implement API calls
    - [x] clean up UI organization
    - [x] styling
- note: Not using AdonisJS because it doesn't seem to support async client updates (will discuss)

## Context

- fixes #379
- fixes #488 

## Checklist

- [x] Related issues linked using `fixes #number`
- [ ] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass

### CODE REVIEW

**Review by sabineschaller — COMMENTED** [review]
Please add the grant seed like we discussed.

File: packages/mock-account-provider/app/consent-screen.tsx [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+  }
+
+  public static async endInteraction(
+    params: Record<string, string>
+  ): Promise<ApiResponse> {
+    // end interaction --> GET /interact/:id/:nonce/finish
+    const { interactId, nonce } = params
+    const response = await ApiSteps.apiCall(
+      'GET',
+      `/interact/${interactId}/${nonce}/finish`,
```
Comment (njlie) [inline packages/mock-account-provider/app/consent-screen.tsx:125]: This endpoint returns a redirect, will this call redirect the browser if it gets a `302`?

Comment (dclipp) [reply packages/mock-account-provider/app/consent-screen.tsx:125]: Yeah, the browser will redirect to whatever the `Location` header points to

**Comment by sabineschaller** [comment]
@dclipp after our 1:1, here is what I think this should do:

- a grant is automatically seeded in pending state when the stack is fired up via docker compose
- when I navigate to localhost:3030 (location of your app), this grant's `interactId` and `nonce` are the default values in the form (but I could also input other info if I had e.g. done a grant request via postman)
- when I click "Begin step", I'm redirected to a consent screen (under the hood, the app does the get request to get the grant details)
- I accept or reject on the consent screen (under the hood, the post to the auth server happens)
- I'm redirected to the initial page that shows the grant that was issued or the rejection message

Let me know if that makes sense to you

File: packages/mock-account-provider/app/routes/shoe-shop.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      <div className='row'>
+        <div className='col-12'>
+          <i className='bi bi-check-circle-fill text-success display-1'></i>
+        </div>
+      </div>
+      <div className='row mt-2'>
+        <div className='col-12'>
+          <p>
+            You gave {thirdPartyName} permission to send {currencyDisplayCode}{' '}
+            {amount.toFixed(2)} out of your account.
```
Comment (wilsonianb) [inline packages/mock-account-provider/app/routes/shoe-shop.tsx:30]: Is the asset scale being applied to the amount here?
On the consent screen, I see:
>You gave undefined permission to send USD 2850.00 out of your account.

For a quote sendAmount of `57.00`

Comment (sabineschaller) [reply packages/mock-account-provider/app/routes/shoe-shop.tsx:30]: Just fixed that

Comment (wilsonianb) [reply packages/mock-account-provider/app/routes/shoe-shop.tsx:30]: I guess we're also missing the `thirdPartyName`
(I'm seeing undefined)

**Review by sabineschaller — APPROVED** [review]
Let's clean this up later

**Pull request merged**